crates/ide_assists/src/handlers/generate_default_from_new.rs

```diff
@@ -0,0 +1,230 @@
+use crate::{
+    assist_context::{AssistContext, Assists},
+    AssistId,
+};
+use syntax::{
+    ast::{self, NameOwner},
+    AstNode, SyntaxKind, SyntaxNode, SyntaxText,
+};
+use test_utils::mark;
+
+// Assist: generate_default_from_new
+//
+// Generates default implementation from new method.
+//
+// ```
+// struct Example { _inner: () }
+//
+// impl Example {
+//     pub fn n$0ew() -> Self {
+//         Self { _inner: () }
+//     }
+// }
+// ```
+// ->
+// ```
+// struct Example { _inner: () }
+//
+// impl Example {
+//     pub fn new() -> Self {
+//         Self { _inner: () }
+//     }
+// }
+//
+// impl Default for Example {
+//     fn default() -> Self {
+//         Self::new()
+//     }
+// }
+// ```
+pub(crate) fn generate_default_from_new(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let fn_node: ast::Fn = ctx.find_node_at_offset()?;
+    let fn_name = fn_node.name()?.to_string();
+
+    if !fn_name.eq("new") {
+        mark::hit!(other_function_than_new);
+        return None;
+    }
+
+    if fn_node.param_list()?.params().count() != 0 {
+        mark::hit!(new_function_with_parameters);
+        return None;
+    }
+
+    let insert_after = scope_for_fn_insertion_node(&fn_node.syntax())?;
+    let impl_obj = ast::Impl::cast(insert_after)?;
```

Comment:
```suggestion
	let impl_ = node.ancestors().into_iter().find_map(ast::Impl::cast)?;
```
No need for the extra function as its only used once